pkg/actions/kv_run_results_iterator.go

```diff
@@ -62,18 +64,24 @@ func (i *KVRunResultIterator) Next() bool {
 	if i.Err() != nil {
 		return false
 	}
-	return i.it.Next()
+	if !i.it.Next() {
+		i.entry = nil
+		return false
+	}
+	e := i.it.Entry()
+	if e == nil {
```

Comment:
How can this happen?
Is it only as a result of a bug, or is there any valid scenario where we may populate a nil value?
I mean, given the preceding `i.it.Next()` return `true`

Reply:
Yeah, I have the same feeling that this might be a bit too defensive. It's not the only place though and I kept the existing behaviour as is. Maybe @N-o-Z can comment on that?

Reply:
We do not allow nil value in the Store - so this should never happen